### ISSUE TITLE
Newly created file from CLI: intellisense only work after first saving

### ISSUE THREAD

**Issue by quantonganh** [microsoft/vscode-go#2760]
- vscode 1.38.1
- ms-vscode.Go@0.11.4

I'm not using the language server.

Steps to Reproduce:
1. Run `code hello.go` from CLI
2. Type the following:

```
package main

import "fmt"

func main() {
    fmt.
```

Nothing shown because `gocode` is not running. But if I save this file first, delete dot and type again, it will be ok:

```
  501 19958     1   0  1:54PM ??         0:00.07 /Users/quanta/go/bin/gocode -s -sock unix -addr 127.0.0.1:37373
```

`gocode` is triggered to run after saving?

I can also make it works by opening the current folder (`code .`), then create a new file from there.

**Comment by ramya-rao-a** [comment]
I am not able to replicate this... Below are the steps I took

- Create a file `hello.go` with the below contents
```go
package main
import "fmt"
func main() {
	fmt
```
- Run `code hello.go` from cli
- Type `.` after `fmt` in the main() function

I am able to see the completions

Did I miss any steps?

**Comment by quantonganh** [comment]
@ramya-rao-a You mean you used another editor to create `hello.go` first, then open it with `code`? That isn't what I mean.

What I mean is creating a new file from cli by using `code hello.go`, then I can only see the completions after first saving.

**Comment by ramya-rao-a** [comment]
> What I mean is creating a new file from cli by using code hello.go, then I can only see the completions after first saving.

So after creating a new file, how did you add the contents? Did you type them in or copy/paste them from another location?

**Comment by quantonganh** [comment]
> Did you type them in or copy/paste them from another location?

I typed it directly.

**Comment by ramya-rao-a** [comment]
Apologies for the late reply

`gocode` requires a saved file to act on. That is the reason you were not able to see any completions.

Interestingly, `gopls` does not have the same restrictions and I was able to get completions for unsaved file too when using the language server.

Closing this issue as we don't have plans to make completions work with an unsaved file in this manner.

Thanks for reporting & Happy Coding!
